pkg/http/httpserver/errorresp/errors.go

```diff
@@ -117,6 +129,13 @@ func (e *APIError) appendLocaleTemplate(template *i18n.Template, args ...interfa
 	return e
 }
 
+func (e *APIError) appendCtx(ctx interface{}) *APIError {
+	if e.ctx == nil {
+		e.ctx = ctx
+	}
+	return e
+}
+
```

Comment:
you mean if e.ctx equals to nil, you do nothing, right? can you tell me, what does `append` mean?